### DISCUSSION TITLE
#xcm Trying to encode native token as Here fails with AssetHasNoReserve in XToken transfer

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [open-web3-stack/open-runtime-module-library#820] [Q&A] [answered]
If I encode my native token as `(1, para id, general key)` XTokens can send it to other chain.

If I encode my native token as `(0, here)` XTokens fails to send it to other chain with AssetHasNoReserve on sender side (so XCM never send and tx fails - so it is not receiver chain issue).

Parity pallet-xcm sends `(0,here)` encoded token and other chain reseives..

I try to encode as `(0, Here)` as I see that is common locally to encode `(0, here)` so it is transformed by executor and reaches other chain as `(1, para id)`

**Comment by dzmitry-lahoda** [comment]
Also I see weird with non native. Like it is gated somehow and I get `NotCrossChainTransferableCurrency` . Also It can be encoded into Origin ok.  Because RelayerXcm::limited_reserve_transfer_assets sends fine. 

So same configuration of XCM, some possibly valid imho configs work with RelayerXcm, but no XTokens.

**Reply by dzmitry-lahoda** [reply]
I could have made some misconfiguration of xTokens, not sure what so :(

**Comment by shaunxw** [comment, marked as answer]
It looks to me that you misconfigured xtokens. It might be the `SelfLocation` and `ReserveProvider` in your config is not consistent. If either of them is configured with interpreting locations as relative view(with `Here` junction), the other one should be relative as well. You may refer to this config in unit tests mock as an example: https://github.com/open-web3-stack/open-runtime-module-library/blob/2c48b626addafbea0227c068c9ab0fc16666b1b5/xtokens/src/mock/para_relative_view.rs#L319-L334

As for `NotCrossChainTransferableCurrency` error, it raises if you don't config the `MultiLocation` for the currency to transfer in `CurrencyIdConvert`.
